src/main/java/net/openhft/chronicle/wire/TextWire.java

```diff
@@ -1670,6 +1670,7 @@ protected long readLengthMarshallable() {
                 // Reset the consumption flag and reading position
                 this.consumeAny = false;
                 bytes.readPosition(start);
+                sb.setLength(0);
```

Comment:
It looks like the pattern is usually to call `acquireStringBuilder()` which will set the length to zero before returning it to you. Perhaps the problem is someone using `sb` directly rather than attaining it via `acquireStringBuilder()`

A better pattern would be to use `ScopedResource<StringBuilder>` but I don't fancy doing that refactor.

Reply:
In this case, acquireStringBuilder() is called as the first line of TextWire#text - the readLength method then uses the underlying StringBuilder as part of consuming content to determine length.

So if I'm understanding correctly, that pattern has been observed, but too early to avoid the impact from readLength.

Reply:
This is actually a prime example of where `ScopedThreadLocal<StringBuilder>` would eliminate the possibility of/alert you to a bug. The nested context is garbling a resource that the outer context thinks it has "acquired".

I kinda think this change will eliminate a single instance of a bug that's probably repeated all through this class.

Reply:
My vote would be to investigate the use of `ScopedThreadLocal<StringBuilder>` in place of the `sb` field.

Reply:
Can absolutely see where you're coming from, @nicktindall - appreciated!

Reply:
Lets merge this fix but please add a TODO: on that line for ScopedThreadLocal so we can remove that line

Reply:
Done!